yt_idefix/data_structures.py

```diff
@@ -724,3 +797,241 @@ def _validate_units_override_keys(cls, units_override):
+        )
+        with open(grid_file) as gridtxt:
+            txt = gridtxt.readlines()
+            self.domain_left_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[2]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.domain_right_edge = np.array(
+                [
+                    float(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[3]
+                    )
+                    for i in [6, 7, 8]
+                ]
+            )
+            self.dimensionality = int(txt[4][-2])
+            self.domain_dimensions = np.array(
+                [
+                    int(
+                        txt[i]
+                        .replace(",", "")
+                        .replace("[", "")
+                        .replace("]", "")
+                        .split()[4]
+                    )
+                    for i in [6, 7, 8]
+                ][: self.dimensionality]
+            )
+
+            def parse_geometry(geom: str):
+                import yt
+
+                if yt.version_info[:2] > (4, 1):
+                    try:
+                        from yt.geometry.api import (
+                            Geometry,  # type: ignore [attr-defined]
+                        )
+
+                        return Geometry(geom)
+                    except ImportError:
+                        pass
+
+                return geom
+
+            geom_str = (txt[5].split()[-1]).lower()
+            self.geometry = parse_geometry(geom_str)
+
+        with open(out_file) as outttxt:
+            txt = outttxt.readlines()
+            entry = int(
+                os.path.basename(self.parameter_filename)
+                .replace(".flt.h5", "")
+                .replace("data.", "")
+                .replace(".dbl.h5", "")
+            )
+            self.current_time = float(txt[entry].split()[1])
+            self.ntracers = 0
+            search = "tr1"
+            while search in txt[entry].split():
+                self.ntracers += 1
+                search = "tr%d" % (self.ntracers + 1)
+
+        self.fluid_types += (self._dataset_type,)
+        self.gamma = 5.0 / 3.0
+        self.mu = 0.61
+        self.storage_filename = self.parameter_filename
+        self.refine_by = 1  # no mesh refinement
+        self._periodicity = (True, True, True)
+        # PlutoStatic cannot yet be run as a cosmological simulation
+        self.cosmological_simulation = 0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
```

Comment:
You'd get this for free by calling the parent class as I mentioned in a previous comment

Reply:
```IdefixDataset._parse_parameter_file(self)``` has other function calls like 
```
self._parse_inifile()
self._parse_definitions_header()
self._setup_geometry()
```
These functions are not there in xdmf reads and going to cause problems. Therefore, I avoided calling the inherited ```IdefixDataset._parse_parameter_file(self)```.

Reply:
I think this has now been addressed.